createManagedMasterK8s.groovy

```diff
@@ -75,19 +83,24 @@ String  k8sYaml = ""
 /**
  * cascBundle (optional). Configuration as Code Configuration Bundle
  */
- String cascBundle = ""
+String cascBundle = ""
 
 /*****************
- * CREATE MASTER *
+ * CREATE CONTROLLER *
  *****************/
-
+def jenkins = Jenkins.get()
+ModifiableTopLevelItemGroup parent = controllerParent?.trim() ? jenkins.getItemByFullName(controllerParent) : jenkins
+if (!parent) {
+    println("Cannot find parent '${controllerParent}'.")
+    return
+}
 /**
- * Create a Managed Masters with just a name (this will automatically fill required values for id, idName, grantId, etc...)
+ * Create a Managed Controllers with just a name (this will automatically fill required values for id, idName, grantId, etc...)
  * Similar to creating an iten in the UI
  */
-ManagedMaster newInstance = jenkins.model.Jenkins.instanceOrNull.createProject(ManagedMaster.class, masterName)
-newInstance.setDescription(masterDescription)
-newInstance.setDisplayName(masterDisplayName)
+ManagedMaster newInstance = parent.createProject(jenkins.getDescriptorByType(ManagedMaster.DescriptorImpl.class), controllerName, true)
```

Comment:
What's this line change do?

Reply:
Lines, 33,91-101 are what implements the option to create a controller inside the folder.
Common interface between `Jenkins` and `Folder` is `ModifiableTopLevelItemGroup`. `ModifiableTopLevelItemGroup` does not have the same `createProject` method as `Jenkins` so I had to change the arguments to make sure it works with the `Folder` as well.

Reply:
Cool. Thanks for the explanation.